index.js

```diff
@@ -13,13 +16,21 @@ const Request = {
     c.uint.preencode(state, req.fork)
     c.fixed32.preencode(state, req.treeHash)
     m.manifest.preencode(state, req.manifest)
+    c.uint8.preencode(state, 0)
+    if (req.blobs === null) return
+    c.uint.preencode(state, req.blobs.length)
+    c.fixed32.preencode(state, req.blobs.treeHash)
   },
   encode (state, req) {
     c.uint.encode(state, req.version)
     c.uint.encode(state, req.length)
     c.uint.encode(state, req.fork)
     c.fixed32.encode(state, req.treeHash)
     m.manifest.encode(state, req.manifest)
+    c.uint.encode(state, req.blobs ? 1 : 0)
```

Comment:
This is a c.uint8 in the preencode step.

Optional: depending on the exact intent, this could also be modeled using the flag-byte pattern (I think that generalises easier if we want to add more conditionals later)

```
preencode (state, i) {
    state.end++ // flags
   ...
    if (req.blobs) {
      c.uint.preencode(state, req.blobs.length)
      c.fixed32.preencode(state, req.blobs.treeHash) 
   }
  },

  encode (state, req) {
      const start = state.start++ // flags
      ...

     let flags = 0
     if (req.blobs) {
        flags |= 1
        c.uint.preencode(state, req.blobs.length)
        c.fixed32.preencode(state, req.blobs.treeHash) 
    }    
    state.buffer[start] = flags
  }
   
   decode (state, req) {
      const flags = c.uint.decode(state)
      const isDrive = flags & 1
      const blobs = isDrive
          ...
       
     }
   }
```